apps/homescreen/js/app.js

```diff
@@ -726,6 +748,97 @@
       return this.icons.children.indexOf(child);
     },
 
+    removeSelectedIcon: function() {
+      if (!this.selectedIcon) {
+        return;
+      }
+
+      if (this.selectedIcon.app) {
+        if (!this.selectedIcon.app.removable) {
+          return;
+        }
+        navigator.mozApps.mgmt.uninstall(this.selectedIcon.app);
+      } else if (this.selectedIcon.bookmark) {
+        new MozActivity({
+          name: 'remove-bookmark',
+          data: { type: 'url', url: this.selectedIcon.bookmark.id }
+        });
+      }
+    },
+
+    editSelectedIcon: function() {
+      if (!this.selectedIcon || !this.selectedIcon.bookmark) {
+        return;
+      }
+
+      new MozActivity({
+        name: 'save-bookmark',
+        data: { type: 'url', url: this.selectedIcon.bookmark.id }
+      });
+    },
+
+    iconIsEditable: function(icon) {
```

Comment:
`iconIsEditable` means an icon can be edited (renamed) and removed? The naming is a bit confusing with `editSelectedIcon` above.
I suggest renaming `editSelectedIcon` to `renameSelectedIcon`. Edit will refer to both renaming and removing. What do  you think?